docs/design_proposals/multi_environment_proposal.md

```diff
@@ -0,0 +1,277 @@
+# Proposal Design: Multi Environment Support
+## Objective
+The aim is to introduce an environment set mechanism in the `pixi` package manager.
+This mechanism will enable clear, conflict-free management of dependencies tailored to specific environments, while also maintaining the integrity of fixed lockfiles.
+
+
+### Motivating Example
+There are multiple scenarios where multiple environments are useful.
+
+- **Testing of multiple package versions**, e.g. `py39` and `py310` or polars `0.12` and `0.13`.
+- **Smaller single tool environments**, e.g. `lint` or `docs`.
+- **Large developer environments**, that combine all the smaller environments, e.g. `dev`.
+- **Strict supersets of environments**, e.g. `prod` and `test-prod` where `test-prod` is a strict superset of `prod`.
+- **Multiple machines from one project**, e.g. a `cuda` environment and a `cpu` environment.
+- **And many more.** (If you have a use-case please add it to the list, so we can make sure it's covered)
+
+This prepares `pixi` for the use in large projects with multiple use-cases, multiple developers and different CI needs.
+
+## Design Considerations
+1. **User-friendliness**: Pixi is a user focussed tool this goes beyond developers. The feature should have good error reporting and helpful documentation from the start. This is opinionated so the user sided PR's should be checked by multiple developers.
```

Comment:
What's "user sided PR's" seems like a typo.

Reply:
```suggestion
1. **User-friendliness**: Pixi is a user focussed tool this goes beyond developers. The feature should have good error reporting and helpful documentation from the start.
```
This was a typical Ruben line... I see this wouldn't help so lets remove it.

Reply:
`s/tool this goes beyond/tool that goes beyond/` 😄